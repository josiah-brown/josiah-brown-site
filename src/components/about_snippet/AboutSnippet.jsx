import React from "react";
import { BsArrowUpLeft } from "react-icons/bs";
import { Link } from "react-router-dom";

const AboutSnippet = ({ heading, content, url }) => {
  let head = heading ? heading : "No Heading...";
  let text = content ? content : "No Content...";
  let linked = url ? true : false;

  return (
    <div
      data-testid="about-snippet"
      className="fixed z-50 left-5 md:left-20 lg:left-28 xl:left-40 top-6 flex flex-col md:top-12 text-black dark:text-white"
    >
      <h1 className="text-2xl font-semibold w-full md:text-5xl">
        <Link to="/" className="cursor-pointer" rel="canonical">
          {head}
        </Link>
      </h1>
      <h3 className="text-sm font-extralight w-1/2 md:text-xl">
        {linked ? (
          <Link to={url} className="cursor-pointer md:hover:italic">
            <BsArrowUpLeft className="inline" />
            {" " + text}
          </Link>
        ) : (
          text
        )}
      </h3>
    </div>
  );
};

export default AboutSnippet;
