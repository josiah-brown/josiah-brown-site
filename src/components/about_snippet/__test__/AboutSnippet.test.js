import { render, cleanup } from "@testing-library/react";
import renderer from "react-test-renderer";
import AboutSnippet from "../AboutSnippet";

afterEach(cleanup);
test("renders about snippet without crashing", () => {
  render(<AboutSnippet />);
});

test("renders about snippet heading correctly", () => {
  const { getByTestId } = render(<AboutSnippet heading="heading" />);
  expect(getByTestId("about-snippet")).toHaveTextContent("heading");
});

test("renders about snippet content correctly", () => {
  const { getByTestId } = render(
    <AboutSnippet heading="heading" content="testing about snippet" />
  );
  expect(getByTestId("about-snippet")).toHaveTextContent(
    "testing about snippet"
  );
});

test("matches about snippet snapshot 1", () => {
  const tree = renderer.create(<AboutSnippet content="snapshot 1" />).toJSON();
  expect(tree).toMatchSnapshot();
});

test("matches about snippet snapshot 2", () => {
  const tree = renderer
    .create(<AboutSnippet heading="blah blah blah" content="" />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
