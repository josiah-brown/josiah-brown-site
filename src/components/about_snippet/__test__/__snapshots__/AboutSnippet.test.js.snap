// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches about snippet snapshot 1 1`] = `
<div
  className="fixed z-50 left-5 md:left-20 lg:left-28 xl:left-40 top-6 flex flex-col md:top-12 text-black dark:text-white"
  data-testid="about-snippet"
>
  <h1
    className="text-2xl font-semibold w-full md:text-5xl"
  >
    <a
      className="cursor-pointer"
      href="/"
    >
      No Heading...
    </a>
  </h1>
  <h3
    className="text-sm font-extralight w-1/2 md:text-xl"
  >
    snapshot 1
  </h3>
</div>
`;

exports[`matches about snippet snapshot 2 1`] = `
<div
  className="fixed z-50 left-5 md:left-20 lg:left-28 xl:left-40 top-6 flex flex-col md:top-12 text-black dark:text-white"
  data-testid="about-snippet"
>
  <h1
    className="text-2xl font-semibold w-full md:text-5xl"
  >
    <a
      className="cursor-pointer"
      href="/"
    >
      blah blah blah
    </a>
  </h1>
  <h3
    className="text-sm font-extralight w-1/2 md:text-xl"
  >
    No Content...
  </h3>
</div>
`;
